### PR TITLE
feature -> develop | [FLUTTER] Updated: core directory

### PR DESCRIPTION
**Pull request by we-prajapati-c001**
### [Task#1084 - Documentation For VaahFlutter](https://team.webreinvent.com/backend/collab/project/vaahcms/tasks/209)

Updated core directory

Time invested: hh:mm format | Billable: 27:51 | Non-Billable: 00:00
Billable time is the total time invested till [Task-1084-Update-directory-structure-vaahextendflutter-widgets](https://github.com/webreinvent/vaah-docs/tree/feature/Task-1084-Update-directory-structure-vaahextendflutter-widgets)

Task Video Proof: https://youtu.be/fY1bNxFu3uc

### CODE REVIEW

File: docs/vaahflutter/core/api.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+To use interceptors, Base Api Url, Cookie jar, etc, Api needs to be initialized first before use, so we initialize it using `Api.init()` in [BaseController](../app.md/#base-controller)
+
+```dart
+Api.init();
+```
+
+## Ajax method
+
+- ajax method automatically converts objects from `snake_case To lowerCamelCase` when `Receiving the response`.
```
Comment (themodernpk) [inline docs/vaahflutter/core/api.md:39]: @we-prajapati-c001 keep first letter as Capital

File: docs/vaahflutter/core/api.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+- Parameter name: callback
+- Required: no
+- Type: `Future<void> Function(dynamic data, Response<dynamic>? res)?`
+- Description: callback function
+- Note: returns a null value when an error occurs, otherwise returns the value for the `data` key and whole response.
+- Example:
+```dart{3}
+Api.ajax(
+    url: '/api/users',
+    callback: (data, resp) async { Console.info('data', data); },
```
Comment (themodernpk) [inline docs/vaahflutter/core/api.md:66]: @we-prajapati-c001 write 2 example 1. as inline `callback` as you have already written, 2. call back as new method eg. `sign` & `signAfter` as `callback`

**Review by themodernpk — CHANGES_REQUESTED** [review]

File: docs/vaahflutter/core/api.md (hunk 1 above)
Comment (we-prajapati-c001) [reply docs/vaahflutter/core/api.md]: I've updated this and pushed a new commit.

File: docs/vaahflutter/core/api.md (hunk 2 above)
Comment (we-prajapati-c001) [reply docs/vaahflutter/core/api.md]: I've updated this and pushed a new commit.

**Pull request merged**